src/frequenz/sdk/microgrid/_graph.py

```diff
@@ -266,6 +266,29 @@ def is_chp_chain(self, component: Component) -> bool:
             Whether the specified component is part of a CHP chain.
         """
 
+    @abstractmethod
+    def dfs(
+        self,
+        current_node: Component,
+        visited: Set[Component],
+        condition: Callable[[Component], bool],
+    ) -> Set[Component]:
+        """
+        Search for components that fulfill the condition in the Graph.
+
+        DFS is used for searching the graph. The graph travarsal is stopped
+        once a component fulfills the condition.
+
+        Args:
+            current_node: The current node to search from.
+            visited: The set of visited nodes.
+            condition: The condition function to check for.
+
+        Returns:
+            A set of component ids where the coresponding components fulfill
```

Comment:
s/coresponding/corresponding

Reply:
I'll add an extra commit for fixing the typos!